### PR TITLE
Add unit test infrastructure

### PR DESCRIPTION
**Pull request by nshuba**
Task/Issue URL: https://app.asana.com/0/1202279501986195/1203126514402189/f

### Description
Added basic unit test infrastructure

### Steps to test this PR
See https://github.com/duckduckgo/Android/pull/2611

### CODE REVIEW

File: unit_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Unit Testing Netguard
+
+## Requirements
+An Android x86/64 emulator with no Google Play
+
+## Running Tests
+* Build the app or just the `vpn-network-imp` module
```
Comment (aitorvs) [inline unit_tests/README.md:7]: Typo
```suggestion
* Build the app or just the `vpn-network-impl` module
```

File: unit_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* Execute the following:
+```
+cd android-app/vpn-network/vpn-network-impl/build/intermediates/cmake/debug/obj
+adb push x86_64/ /data/
```
Comment (aitorvs) [inline unit_tests/README.md:11]: This step fails on emulator with `permission denied`. Are we sure we don't need a rooted device?

File: unit_tests/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Unit Testing Netguard
+
+## Requirements
+An Android x86/64 emulator with no Google Play
+
+## Running Tests
+* Build the app or just the `vpn-network-imp` module
+* Execute the following:
+```
+cd android-app/vpn-network/vpn-network-impl/build/intermediates/cmake/debug/obj
```
Comment (aitorvs) [inline unit_tests/README.md:10]: This absolute path only exist in your PC, Maybe is best to change to something like
```suggestion
cd ..../vpn-network-impl/build/intermediates/cmake/debug/obj
```

**Review by aitorvs — COMMENTED** [review]
Can't test this at the minute :/

File: unit_tests/README.md (hunk 1 above)
Comment (aitorvs) [inline unit_tests/README.md:7]: The `test_tcp` is not compiled.
I have tried `./gradlew :vpn-network-impl:assemble`.
The `libnetguard.so` library is in there, not the test executable

**Comment by nshuba** [comment]
Closing this as we moved to the method in https://github.com/duckduckgo/netguard/pull/14 instead.

**Pull request closed without merging**